eng/build.ps1

```diff
@@ -130,6 +130,13 @@ if ($help) {
   exit 0
 }
 
+# check the first argument if subset is not explicitly passed in
+if (-not $PSBoundParameters.ContainsKey("subset") -and $properties.Length -gt 0 -and $properties[0] -match '^[a-zA-Z.+]+$') {
```

Comment:
'^[a-zA-Z.+]+$'

The '.' currently means at least one character of anything. I suspect you intended to escape it to consider a valid character. Right?:

     '^[a-zA-Z\.+]+$'

Are there other characters that could be accepted in properties, like underscore `_` or dash (escaped) `\-`?

Reply:
Actually, the `+` sign also needs escaping if indicated inside square brackets:

    '^[a-zA-Z\.\+]+$'

Reply:
Confirmed - No `_`, `\-` or numbers (`0-9`) needed. Only the escaped `\.` and the `\+` signed: https://github.com/dotnet/runtime/blob/e99836a899bdb97be5afbbed8d90573ee5635b30/eng/Subsets.props#L123-L189

Reply:
no I explicitly tried this (we have the same regex in the .sh) and characters inside character classes don't need to be escaped:

<img width="387" alt="image" src="https://github.com/dotnet/runtime/assets/1376924/95811cad-cbab-4e8d-b396-a1bbc36b176b">

It doesn't hurt to escape them though so it's fine :)